docs/src/components/App/Header/Header.js

```diff
@@ -1,18 +1,81 @@
 import styles from './Header.less';
 
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 
-import { Link } from 'react-router-dom';
-import { PageBlock, Card, Section, Text, ScreenReaderOnly } from 'seek-style-guide/react';
+import { Link, withRouter } from 'react-router-dom';
+import {
+  PageBlock,
+  Card,
+  Highlight,
+  Section,
+  ChevronIcon,
+  Text,
+  TextField,
+  ScreenReaderOnly
+} from 'seek-style-guide/react';
 import Logo from './Logo/Logo';
 import demoSpecExports from '../../../../../react/*/*.demo.js';
 const demoSpecs = demoSpecExports.map(x => x.default);
+const allRoutes = [
+  { route: '/typography', title: 'Typography', category: 'Guides' },
+  { route: '/page-layout', title: 'Page Layout', category: 'Guides' },
+  { route: '/sandbox', title: 'Sandbox', category: 'Tools' },
+  ...demoSpecs
+];
 
-export default class Header extends Component {
+const SEARCH_BAR_ID = 'search-bar-field';
+
+const WithHighlighting = ({ highlighted, children }) => {
+  if (highlighted) {
+    return (
+      <Highlight tone="neutral">
+        <div className={styles.highlightWrapper}>
+          <ChevronIcon className={styles.highlightedIcon} />
+          {children}
+        </div>
+      </Highlight>
+    );
+  }
+
+  return children;
+};
+
+const buildRoutes = inputRoutes =>
+  inputRoutes.reduce((acc, route) => {
+    const category = route.category ? route.category : 'Other';
```

Comment:
If you wanted to, I believe this could be short circuited with:

`const category = route.category || 'Other';`